### PR TITLE
chore: use compatible versions (react/react-native/react-native-web)

### PR DESCRIPTION
**Pull request by fampinheiro**

### CODE REVIEW

**Comment by dickie81** [comment]
So are we going back to previous version on RN?

Is it necessary to have exact versions for peer dependencies? I think this limits our published components reusability.

**Comment by fampinheiro** [comment]
The change to RN `0.42.3` is not the most pleasant change I did in most recent years but it needs to be done because some of the packages that we use (`react-native-web`) the most obvious use private react API and the last RN version that has React 15 on peerDependencies is this.

I don't see a problem in being more permissive on `peerDependencies`.

**Comment by dickie81** [comment]
Can we agree a laxer group of `peerDependecies` to be applied across all React component packages?

Perhaps:

`
peerDependencies: {
    "@storybook/react-native": "3.2.4",
    "react": ">=15",
    "react-dom": ">=15",
    "react-native": ">=0.42"
}`

**Comment by dickie81** [comment]
I'm having problems building android storybook app. Can someone else confirm?

**Comment by fampinheiro** [comment]
@dickie81 https://github.com/react-native-community/react-native-svg/issues/327

**Comment by dickie81** [comment]
https://www.npmjs.com/package/react-native-svg#Instalation

File: lib/cli/package-json.js [hunk 1]
```diff
@@ -40,18 +40,21 @@ module.exports = variables => ({
...
     "prop-types": "15.5.10",
-    react: "16.0.0-alpha.6",
-    "react-dom": "15.5.4",
-    "react-native": "0.44.1",
-    "react-native-web": "0.0.97"
+    "react-native-web": "0.0.119"
   },
   peerDependencies: {
-    "@storybook/react-native": "3.1.6"
+    "@storybook/react-native": "3.2.4",
```
Comment (dickie81) [inline lib/cli/package-json.js]: should these be less strict too?

**Comment by fampinheiro** [comment]
It looks like it was introduced by us.

File: lib/cli/package-json.js [hunk 2]
```diff
@@ -40,18 +40,21 @@ module.exports = variables => ({
...
-    "react-native-web": "0.0.97"
+    "react-native-web": "0.0.119"
   },
   peerDependencies: {
-    "@storybook/react-native": "3.1.6"
+    "@storybook/react-native": ">=3",
+    react: "15.4.2",
+    "react-dom": "15.4.2",
+    "react-native": "0.42.3"
   },
```
Comment (dickie81) [inline lib/cli/package-json.js]: Previous 3 lines too?

**Pull request merged**